usermanagement/src/main/java/com/freshworks/haystack/usermanagement/trigmetry/service/TrigmetryService.java

```
@@ -70,111 +73,138 @@ private Integer getId(String mail) throws Exception {
-      LOG.error("Exception occurred while adding creating global User :" + email, ex);
-      throw new OperationFailException("Operation Failed while adding User:" + email);
-    }
-    LOG.info("Global user added successfully");
-    return true;
-  }
-
-  private boolean globalUserExists(String email) {
-    try {
-      Integer id = this.getId(email);
-      if (null == id) {
-        LOG.info("Global user does not exists with email : " + email);
-        return false;
-      }
-      LOG.info("Global user exists with Id : " + id);
-      return true;
+      LOG.info("Creating new user :" + name);
+      grafanaUserCretationReponse = thirdpartyApiConnector.postCall(url, new StringEntity(payload, StandardCharsets.UTF_8), GrafanaUserCretationReponse.class, getTargetHost(), getContext());
     }
     catch (Exception ex) {
-      LOG.error("Exception occured while looking for email:" + email, ex);
+      LOG.error("Exception occurred while creating User :" + email, ex);
+      return null;
     }
-    return false;
+    LOG.info("User created successfully");
+    return grafanaUserCretationReponse;
   }
 
-  public boolean execute(TrigmetryRequest trigmetryRequest) throws Exception {
+  public String execute(TrigmetryRequest trigmetryRequest) throws Exception {
     String email = trigmetryRequest.getEmail();
+    String userName = this.getUserNameFromEmail(email);
     String orgName = trigmetryRequest.getOrgName();
     String role = trigmetryRequest.getRole();
 
-    if (!this.globalUserExists(email)) {
-      String userName = this.getUserNameFromEmail(email);
-      this.addGlobalUser(userName, email, userName, userName);
+    String responseMessage = "success";
+    int orgId;
+    Integer userId = null;
+    try {
+      LOG.info("Finding Org by name: " + orgName);
+      orgId = this.findOrgIdByName(orgName);
+      LOG.info("Org found with OrgId: " + orgId);
     }
-    if (null == orgName) {
-      return true;
+    catch (Exception ex) {
```

Catching `Exception` will catch all exception. Do we need this here? Can we catch specific exceptions and will catching specific exceptions have any impact? Please check this at other places too.

---

Changed to specific exception whatever code comes under this pr